contracts/MerkleDistributorWithDeadline.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity =0.8.15;
+
+import "./MerkleDistributor.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import {IERC20, SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract MerkleDistributorWithDeadline is MerkleDistributor, Ownable {
+    using SafeERC20 for IERC20;
+
+    uint256 public immutable endTime;
+
+    constructor (address token_, bytes32 merkleRoot_, uint256 endTime_) MerkleDistributor (token_, merkleRoot_) {
+        require(endTime_ > block.timestamp, "endTime_ must be in the future");
```

Comment:
would it at all be beneficial to have a start time? so there would be more like a claim window?